manifests/init.pp

```diff
@@ -314,6 +339,18 @@
   if $purge_config { fail('purge_config is deprecated, set the purge parameter to a hash containing `config => true` instead') }
   if $purge_plugins_dir { fail('purge_plugins_dir is deprecated, set the purge parameter to a hash containing `plugins => true` instead') }
 
+  # sensu-enterprise supercedes sensu-server and sensu-api
+  if ( $enterprise and $api ) or ( $enterprise and $server ) {
+    fail('Sensu Enterprise: sensu-enterprise replaces sensu-server and sensu-api')
+  }
+  # validate enterprise repo creds
+  if $enterprise or $enterprise_dashboard {
+    validate_string($enterprise_user, $enterprise_pass)
+    if $enterprise_user == undef or $enterprise_pass == undef {
```

Comment:
Just a thought on this, many organizations do not allow direct access to remote repos through their firewalls so they may be mirroring the packages internally.  They can get around it by setting these settings to junk strings, just a thought though.  Maybe only require these if install_repos is true?